In [4]:
import pandas as pd
dy1 = pd.read_csv("/Users/noahschutz/Documents/yumstock.csv")
print(dy1)


            date   open     high      low  close  sharetrade Unnamed: 6
0     2013-02-08  63.70  65.4900  63.6600  65.30     7190051        YUM
1     2013-02-11  65.16  65.1900  64.4800  64.55     5352531        YUM
2     2013-02-12  64.58  65.0600  64.0600  64.75     4934311        YUM
3     2013-02-13  64.77  64.9100  63.9500  64.41     3999959        YUM
4     2013-02-14  64.27  64.5360  63.7700  63.89     3949356        YUM
5     2013-02-15  63.82  64.5000  63.4450  63.99     4334641        YUM
6     2013-02-19  64.20  65.1850  63.9300  65.15     4288462        YUM
7     2013-02-20  65.17  65.4600  64.5200  64.56     4302390        YUM
8     2013-02-21  64.27  65.4000  64.2500  65.05     4490422        YUM
9     2013-02-22  65.21  65.5960  64.7200  65.45     2579023        YUM
10    2013-02-25  65.62  66.3900  64.7300  64.73     4389162        YUM
11    2013-02-26  65.15  65.4800  64.3050  64.87     3758313        YUM
12    2013-02-27  64.85  65.6500  64.7600  65.38     1984721    

## So we have our information, but now we want to narrow our information.


In [5]:
dy1


date   open     high      low  close  sharetrade Unnamed: 6
0     2013-02-08  63.70  65.4900  63.6600  65.30     7190051        YUM
1     2013-02-11  65.16  65.1900  64.4800  64.55     5352531        YUM
2     2013-02-12  64.58  65.0600  64.0600  64.75     4934311        YUM
3     2013-02-13  64.77  64.9100  63.9500  64.41     3999959        YUM
4     2013-02-14  64.27  64.5360  63.7700  63.89     3949356        YUM
5     2013-02-15  63.82  64.5000  63.4450  63.99     4334641        YUM
6     2013-02-19  64.20  65.1850  63.9300  65.15     4288462        YUM
7     2013-02-20  65.17  65.4600  64.5200  64.56     4302390        YUM
8     2013-02-21  64.27  65.4000  64.2500  65.05     4490422        YUM
9     2013-02-22  65.21  65.5960  64.7200  65.45     2579023        YUM
10    2013-02-25  65.62  66.3900  64.7300  64.73     4389162        YUM
11    2013-02-26  65.15  65.4800  64.3050  64.87     3758313        YUM
12    2013-02-27  64.85  65.6500  64.7600  65.38     1984721        YUM
13    2013-02-28  65.63  66.4800  65.4300  65.48     3368170        YUM
14    2013-03-01  64.89  65.6200  64.3500  65.21     3079626        YUM
15    2013-03-04  65.11  65.4800  64.9510  65.47     2150226        YUM
16    2013-03-05  65.59  66.8300  65.5500  66.30     3145755        YUM
17    2013-03-06  66.36  68.0000  66.3000  66.91     4263982        YUM
18    2013-03-07  66.82  67.3100  66.6260  67.25     3538003        YUM
19    2013-03-08  67.71  68.3700  67.6700  67.72     4603358        YUM
20    2013-03-11  67.78  67.9700  67.3250  67.84     4619031        YUM
21    2013-03-12  71.39  71.5000  68.5500  68.73     9455646        YUM
22    2013-03-13  68.61  69.6300  68.5500  69.35     5972245        YUM
23    2013-03-14  69.46  69.9100  69.2500  69.85     2674654        YUM
24    2013-03-15  69.53  70.2000  69.2400  70.20     3498751        YUM
25    2013-03-18  69.31  69.6300  68.7500  69.11     3186987        YUM
26    2013-03-19  69.27  69.5400  68.4600  68.84     2390640        YUM
27    2013-03-20  69.18  69.7100  68.9500  69.45     1649075        YUM
28    2013-03-21  69.06  69.9000  69.0600  69.46     1829652        YUM
29    2013-03-22  69.36  69.9000  69.3000  69.69     1777711        YUM
...          ...    ...      ...      ...    ...         ...        ...
1229  2017-12-26  82.39  82.4400  81.9700  82.19      666705        YUM
1230  2017-12-27  82.34  82.5600  81.7000  82.40     1051921        YUM
1231  2017-12-28  82.69  82.6900  82.0200  82.67      673301        YUM
1232  2017-12-29  82.64  82.7100  81.5900  81.61     1347613        YUM
1233  2018-01-02  82.15  82.4600  81.4800  81.60     1747813        YUM
1234  2018-01-03  81.23  81.8200  81.1343  81.53     2554926        YUM
1235  2018-01-04  81.75  82.5900  81.5300  82.36     1971969        YUM
1236  2018-01-05  82.59  83.0100  82.3600  82.84     1927917        YUM
1237  2018-01-08  82.72  83.0600  82.1200  82.98     1599534        YUM
1238  2018-01-09  83.07  83.1599  82.2700  82.76     1701081        YUM
1239  2018-01-10  82.85  82.8500  82.2100  82.55      887313        YUM
1240  2018-01-11  82.58  83.0100  81.9350  82.49     1352675        YUM
1241  2018-01-12  82.76  83.7600  82.4500  83.62     1377469        YUM
1242  2018-01-16  84.07  84.3400  82.6400  82.93     1391154        YUM
1243  2018-01-17  83.32  83.5200  82.7600  83.09     1678525        YUM
1244  2018-01-18  83.18  83.4299  82.9300  83.10     1015220        YUM
1245  2018-01-19  83.43  83.9100  83.3400  83.81     1677686        YUM
1246  2018-01-22  83.81  84.9200  83.6700  84.91     1469255        YUM
1247  2018-01-23  84.72  86.3400  84.6627  85.84     2010054        YUM
1248  2018-01-24  85.97  86.1000  84.9500  85.24     1192932        YUM
1249  2018-01-25  85.32  85.8200  84.6200  85.66      929863        YUM
1250  2018-01-26  85.68  86.2400  85.2100  86.20     1332647        YUM
1251  2018-01-29  86.09  86.9300  85.9300  86.48     2303333        YUM
1252  2018-01-30  85.00  85.4800  84.3700  84.59    

In [6]:
dy1["close"].mean()


75.45100873709283

** This means that over the past 5+ years, YUM stock has been valued at about $75.45 in the markets. **

* Honestly though, this was the third dataset I tried to work, as the first two didn't really give any relevant information that I could talk about. 
* Regardless, for this one I can also develop a summary of the available data.

In [12]:
dy1.describe()


open         high          low        close    sharetrade
count  1259.000000  1259.000000  1259.000000  1259.000000  1.259000e+03
mean     75.422099    76.027123    74.844914    75.451009  3.209032e+06
std       7.488521     7.550618     7.427339     7.485003  2.217774e+06
min      59.890000    60.250000    59.570000    59.700000  6.667050e+05
25%      70.435000    70.934950    69.665000    70.225000  1.973342e+06
50%      74.110000    74.610000    73.650000    74.100000  2.665015e+06
75%      80.380000    81.145000    79.620000    80.470000  3.710812e+06
max      94.950000    95.900000    94.150000    94.880000  3.679555e+07

In [15]:
#can we properly search for a day that the stock opened at a particular value?
dy1["low"].count()


1259

#### so there are 1259 dates we can search for

In [23]:
dy1["open"].index[82.15]
#this is where I've gotten a bit stuck, as I've tried multiple combinations of syntax to get this code to work, probably need another command other than index but idk


IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices